tests/type_propagation/gold/nlr_test.gold

```diff
@@ -0,0 +1,143 @@
+main tests/type_propagation/nlr_test.toit
+  0[053] - invoke static test_simple tests/type_propagation/nlr_test.toit // {Null_}
+  3[041] - pop 1
+  4[053] - invoke static test_try tests/type_propagation/nlr_test.toit // {Null_}
+  7[089] - return null S1 0
+
+test_simple tests/type_propagation/nlr_test.toit
+  0[053] - invoke static always_return tests/type_propagation/nlr_test.toit // {SmallInteger_}
+  3[041] - pop 1
+  4[053] - invoke static maybe_return tests/type_propagation/nlr_test.toit // {String_|SmallInteger_}
+  7[089] - return null S1 0
+
+test_try tests/type_propagation/nlr_test.toit
+  0[053] - invoke static stop_unwinding tests/type_propagation/nlr_test.toit // {String_|float|SmallInteger_}
```

Comment:
Would be nice if `SmallInteger_` wasn't part of the set.
The block of a `try` is always executed, and the first two assignments aren't throwing.